tony-history-server/app/utils/ParserUtils.java

```diff
@@ -0,0 +1,88 @@
+package utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.gson.JsonSyntaxException;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import models.JobConfig;
+import models.JobMetadata;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import play.Logger;
+import play.libs.Json;
+
+import static utils.HdfsUtils.*;
+
+/**
+ * The class handles parsing different file format
+ */
+public class ParserUtils {
+  private static final Logger.ALogger LOG = Logger.of(ParserUtils.class);
+
+  public static JobMetadata parseMetadata(FileSystem fs, Path metricsFilePath) {
+    if (!isPathValid(fs, metricsFilePath)) {
+      return new JobMetadata();
+    }
+
+    String fileContent = contentOfHdfsFile(fs, metricsFilePath);
+    JobMetadata jobMetadata = new JobMetadata();
+    JsonNode jObj;
+    try {
+      jObj = Json.parse(fileContent);
+    } catch (JsonSyntaxException e) {
+      LOG.error("Couldn't parse metadata", e);
+      return jobMetadata;
+    }
+    jobMetadata.setId(jObj.get("id").textValue());
+    jobMetadata.setJobLink(jObj.get("url").textValue());
+    jobMetadata.setConfigLink("/jobs/" + jobMetadata.getId());
+    jobMetadata.setStarted(jObj.get("started").textValue());
+    jobMetadata.setCompleted(jObj.get("completed").textValue());
+    LOG.info("Successfully parsed metadata");
```

Comment:
Remove or make .debug

Reply:
So what are the rules/best practices for putting something as .info vs. .debug?

Reply:
👍

Reply:
Fixed: https://github.com/linkedin/TonY/pull/62/commits/00bc6de64cef38b2c35f288a31d5cb99357f3eef

Reply:
INFO is usually the default log level, meaning users will generally see these messages. DEBUG messages won't get shown if the log level is INFO, since DEBUG is at a higher verbosity level. Usually, you only need to enable DEBUG logging while developing / debugging.

You want to keep the number of INFO messages small so the logs aren't flooded with repetitive messages. DEBUG logging can be more verbose since you generally enable it only when you're trying to figure out a bug or implement a new feature.